cpu/native/syscalls.c

```diff
@@ -237,7 +246,28 @@ ssize_t _native_write(int fd, const void *buf, size_t count)
 
 ssize_t _native_writev(int fd, const struct iovec *iov, int iovcnt)
 {
-    ssize_t r;
+    ssize_t r = 0;
+
+    if (fd == STDOUT_FILENO || fd == STDERR_FILENO) {
+        while (iovcnt--) {
+            ssize_t res = stdio_write(iov->iov_base, iov->iov_len);
+
+            if (res >= 0) {
+                r += res;
+            } else {
+                errno = res;
+                return -1;
```

Comment:
`*write` should already have set `errno` -at least in case `real_write` is called (default)- the return value should not be an ERRNO but according to standard is -1 (in case of failure)
```suggestion
                return res;
```

Reply:
`return -1` would also be ok but in that case we need may have another `set return register = -1` instruction